src/slowmokit/methods/metrics/classification_report.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file methods/metrics/classification_report.hpp
+ *
+ * Easy include to show the classification report of data
+ */
+
+#include "classification_report.hpp"
+template <class T>
+
+ClassificationReport<T>::ClassificationReport(std::vector<T> &trueValue, std::vector<T> &predictedValue)
+{
+    this->trueValue = trueValue;
+    this->predictedValue = predictedValue;
+}
+
+template <class T>
+void ClassificationReport<T>::confusionMatrix(std::vector<T> &trueValue, std::vector<T> &predictedValue)
+{
+    int n = trueValue.size();
+
+    for (int i = 0; i < n; i++)
+    {
+        classes[trueValue[i]]++;
+        if (trueValue[i] == predictedValue[i])
+        {
+            truePositive[trueValue[i]]++;
+        }
+        if (trueValue[i] != predictedValue[i])
+        {
+            falseNegative[trueValue[i]]++;
+            falsePositive[predictedValue[i]]++;
+        }
+    }
+    // For True Negative, had to do it separately :crycatok:
+    for (auto it : classes)
+    {
+        int classNumber = it.first;
```

Comment:
type of it.first is T and not int. This is done at some other places too....see and correct it